src/lib/logs/gcp_client.py

```diff
@@ -0,0 +1,83 @@
+#   Copyright 2024 Dynatrace LLC
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import asyncio
+import json
+from typing import Any, Dict, List
+
+from lib.clientsession_provider import init_gcp_client_session
+from lib.context import LoggingContext
+from lib.credentials import create_token
+from lib.logs.log_forwarder_variables import (
+    LOGS_SUBSCRIPTION_ID,
+    LOGS_SUBSCRIPTION_PROJECT,
+    PROCESSING_WORKER_PULL_REQUEST_MAX_MESSAGES,
+)
+
+SUBSCRIPTION_PATH = f"projects/{LOGS_SUBSCRIPTION_PROJECT}/subscriptions/{LOGS_SUBSCRIPTION_ID}"
+
+
+class GCPClient:
+    subscription_path: str
+    pull_url: str
+    acknowledge_url: str
+    body_payload: bytes
+    headers: Dict[str, Any]
+
+    def __init__(
+        self,
+        api_token: str,
+    ):
+        self.subscription_path = (
+            f"projects/{LOGS_SUBSCRIPTION_PROJECT}/subscriptions/{LOGS_SUBSCRIPTION_ID}"
+        )
+        self.pull_url = f"https://pubsub.googleapis.com/v1/{SUBSCRIPTION_PATH}:pull"
+        self.acknowledge_url = f"https://pubsub.googleapis.com/v1/{SUBSCRIPTION_PATH}:acknowledge"
+        self.headers = {"Authorization": f"Bearer {api_token}"}
+        json_body = {"maxMessages": PROCESSING_WORKER_PULL_REQUEST_MAX_MESSAGES}
+        json_data = json.dumps(json_body)
+        self.body_payload = json_data.encode("utf-8")
+
+    async def pull_messages(
+        self, logging_context: LoggingContext, gcp_session
+    ) -> Dict[str, List[Any]]:  # type: ignore
+        async with gcp_session.request(
```

Comment:
Hey, just a question - is timing out handled in this request? Do we know it won't hang forever?